ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/rlpx/connections/AbstractPeerConnection.java

```diff
@@ -185,15 +185,18 @@ public void terminateConnection(final DisconnectReason reason, final boolean pee
   @Override
   public void disconnect(final DisconnectReason reason) {
     if (disconnected.compareAndSet(false, true)) {
-      connectionEventDispatcher.dispatchDisconnect(this, reason, false);
-      doSend(null, DisconnectMessage.create(reason));
-      LOG.atDebug()
-          .setMessage("Disconnecting connection {}, peer {} reason {}")
-          .addArgument(this.hashCode())
-          .addArgument(peer.getLoggableId())
-          .addArgument(reason)
-          .log();
-      closeConnection();
+      try {
+        doSend(null, DisconnectMessage.create(reason));
```

Comment:
It would be nice to add a comment saying that we need to do the doSend first because the dispatchDisconnect could fail with an exception